### PR TITLE
feat(ibc): update grpc msg

### PR DESCRIPTION
**Pull request by wenyuanhust**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Which docs this PR relation**:

Ref #

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

### CI Switch

- [x] Chaos CI
- [x] Cargo Clippy
- [x] Coverage Test
- [x] E2E Tests
- [x] Code Format
- [x] Unit Tests
- [x] Web3 Compatible Tests
- [x] OCT 1-5 And 12-15
- [x] OCT 6-10
- [x] OCT 11
- [x] OCT 16-19
- [x] v3 Core Tests

### CODE REVIEW

File: core/ibc/src/grpc.rs [hunk 1]
```diff
@@ -780,74 +839,63 @@ impl<Ctx: ClientReader + ClientKeeper> IbcClientMsgService<Ctx> {
     pub fn new(ctx: Arc<RwLock<Ctx>>) -> Self {
         Self { ctx }
     }
+
+    fn process_msg(&self, msg: ClientMsg) -> Result<ClientResult, tonic::Status> {
+        let mut ctx = self
+            .ctx
+            .write()
+            .map_err(|v| tonic::Status::internal(format!("{:?}", v)))?;
```
Comment (KaoImin) [inline core/ibc/src/grpc.rs]:
```suggestion
            .map_err(|v| tonic::Status::internal(v.to_string()))?;
```

File: core/ibc/src/grpc.rs [hunk 2]
```diff
@@ -780,74 +839,63 @@ impl<Ctx: ClientReader + ClientKeeper> IbcClientMsgService<Ctx> {
...
+        let output = client_dispatch(ctx.deref(), msg)
```
Comment (KaoImin) [inline core/ibc/src/grpc.rs]: Is it necessary to use `WriteGuard` here, or `ReadGuard` also works?

File: core/ibc/src/grpc.rs [hunk 3]
```diff
@@ -53,6 +99,7 @@ use ibc_proto::ibc::core::{
         QueryConnectionsRequest, QueryConnectionsResponse,
     },
 };
+
 use tonic::{transport::Server, Request, Response, Status};
```
Comment (KaoImin) [inline core/ibc/src/grpc.rs:103]: The `Request`, `Response`, and `Status` have imported. Do not quote from crate root.

File: core/ibc/src/grpc.rs (hunk 2 above)
Comment (wenyuanhust) [reply core/ibc/src/grpc.rs]: read is enough for dispatch, but the following store_**_result needs Write, just to simplify the logic

File: core/ibc/src/grpc.rs [hunk 4]
```diff
@@ -780,80 +839,302 @@ impl<Ctx: ClientReader + ClientKeeper> IbcClientMsgService<Ctx> {
...
+            .write()
+            .map_err(|v| Status::internal(v.to_string()))?;
+        let output =
+            client_dispatch(ctx.deref(), msg).map_err(|v| Status::internal(v.to_string()))?;
+        // Apply the result to the context (host chain store).
+        ctx.store_client_result(output.result.clone())
+            .map_err(|v| Status::internal(v.to_string()))?;
+
+        Ok(output.result)
+    }
```
Comment (KaoImin) [inline core/ibc/src/grpc.rs]:
```suggestion
    fn process_msg(&self, msg: ClientMsg) -> Result<ClientResult, Status> {
        {
            let mut ctx = self
                .ctx
                .read()
                .map_err(|v| Status::internal(v.to_string()))?;
            let output =
                client_dispatch(ctx.deref(), msg).map_err(|v| Status::internal(v.to_string()))?;
        }
        // Apply the result to the context (host chain store).
        let mut ctx = self.ctx.write();
        ctx.store_client_result(output.result.clone())
            .map_err(|v| Status::internal(v.to_string()))?;

        Ok(output.result)
    }
```

**Review by KaoImin — APPROVED** [review]

**Pull request merged**